dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/GithubAppCheck.java

```diff
@@ -75,40 +83,70 @@ public GithubAppCheck(final Namespace ns){
      * @return True if github app is installed, false otherwise. 
      */
     protected boolean isGithubAppEnabledOnRepository(String fullRepoName) {
-        Integer maxRetryCount = 5;
-        Boolean isGithubAppInstalled = isGithubAppEnabledOnRepositoryWithGitApi(fullRepoName);
-        if (!isGithubAppInstalled) {
-            for (Integer i = 0; i < maxRetryCount; i++) {
-                isGithubAppInstalled = isGithubAppEnabledOnRepositoryWithGitApi(fullRepoName);
-                if (isGithubAppInstalled) break;
-            }
-        }
-        if (!isGithubAppInstalled) {
-            isGithubAppInstalled = isGithubAppEnabledOnRepositoryWithRenovateApi(fullRepoName);
+        boolean result = isGithubAppEnabledOnRepositoryWithRenovateApi_Retry(fullRepoName);
+        
+        if (!result) {
+            result = isGithubAppEnabledOnRepositoryWithGitApi_Retry(fullRepoName);
         }
-        return isGithubAppInstalled;
+        
+        return result;
+    }
+
+
+    protected boolean isGithubAppEnabledOnRepositoryWithRenovateApi_Retry(String fullRepoName) {
+        RetryConfig config = RetryConfig.custom()
+                .maxAttempts(1)
+                .waitDuration(Duration.ofMillis(1000))
+                .retryExceptions(IOException.class)
+                .build();
+        Retry retry = Retry.of("id", config);
+
+        Try<Boolean> retryResult = Try.ofSupplier(Retry.decorateSupplier(retry, () -> isGithubAppEnabledOnRepositoryWithRenovateApi(fullRepoName)));
+        
+        return retryResult.get();
+    }
+
+
+    protected boolean isGithubAppEnabledOnRepositoryWithGitApi_Retry(String fullRepoName) {
+        RetryConfig config = RetryConfig.custom()
+                .maxAttempts(1)
+                .waitDuration(Duration.ofMillis(1000))
+                .retryExceptions(IOException.class)
+                .build();
+        Retry retry = Retry.of("id", config);
```

Comment:
This still won't retry anything because you are catching all IOExceptions within the isGithubAppEnabledOnRepositoryWithGitApi method.

Reply:
@cjneasbi Fix that with the exception throw with status code 5XX